### PR TITLE
integrate cost centre actor and inject it into workflow

### PR DESCRIPTION
**Pull request by VaclavK**

### CODE REVIEW

File: src/Mammon/Workflows/Activities/AssignCostCentreActivity.cs [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+        ArgumentNullException.ThrowIfNull(input);
+
+        //get cost centre costs
+        var costCentreCosts = await ActorProxy.DefaultProxyFactory.CallActorWithNoTimeout<IResourceActor, IDictionary<string, double>>(input.ResourceActorId, "ResourceActor", async (p) => await p.AssignCostCentreCosts()) 
+            ?? throw new InvalidOperationException($"Unexpected result returned for ${input.ResourceId} - AssignCostCentreCosts");
+
+        foreach (var (costCentre, costCentreCost) in costCentreCosts)
+        {            
+            //send them to cost centre actors
+            await ActorProxy.DefaultProxyFactory.CallActorWithNoTimeout<ICostCentreActor>($"{input.ReportId}_{costCentre}", "CostCentreActor", async (p) => await p.AddCost(input.ResourceId, costCentreCost));
```
Comment (rmjoia) [inline src/Mammon/Workflows/Activities/AssignCostCentreActivity.cs:16]: should we rename this `AddCost` to `AddCostAsync`?

Comment (rmjoia) [reply src/Mammon/Workflows/Activities/AssignCostCentreActivity.cs:16]: and `AssignCostCentreCosts`? maybe if you only have async methods, you don't need to differentiate :| not sure what's the convention

**Review by rmjoia — APPROVED** [review]

File: src/Mammon/Workflows/Activities/AssignCostCentreActivity.cs (hunk 1 above)
Comment (VaclavK) [reply src/Mammon/Workflows/Activities/AssignCostCentreActivity.cs:16]: let me do it... I kinda belive async suffix is artificial (visible from signature)  but lot of people need that to sleep better :D

File: src/Mammon/Properties/launchSettings.json [hunk 2]
```diff
@@ -7,7 +7,8 @@
       "workingDirectory": "$(ProjectDir)",
       "environmentVariables": {
         "ASPNETCORE_URLS": "http://+:8082",
-        "Mammon__MockCostAPIResponseFilePath": "./Services/dummyCostAPIResponse.json"
+        "Mammon__MockCostAPIResponseFilePath": "./Services/dummyCostAPIResponse.json",
+        "Mammon__CostCentreRuleEngineFilePath": "../../../costcentre-definitions/costCentreRules.json"
```
[the remark this answers is not in the document]
Comment (VaclavK) [reply src/Mammon/Properties/launchSettings.json:11]: discussed on the phone - this is pointing to another (private) repo checkout with tenant specific file

**Pull request merged**